sys/hashes/sha256.c

```diff
@@ -223,7 +223,9 @@ void sha256_update(sha256_context_t *ctx, const void *data, size_t len)
 
     /* Handle the case where we don't need to perform any transforms */
     if (len < 64 - r) {
-        memcpy(&ctx->buf[r], data, len);
+        if (len > 0) {
```

Comment:
Just a suggestion but how about pulling up the expression into the `if` above with an `&& (len > 0)`?

Reply:
That changes the semantics.
When len is zero, just return. If ```len < 64 - r```, do the memcpy first.

Reply:
Ah right! Missed the `return` below... Sorry for the noise.